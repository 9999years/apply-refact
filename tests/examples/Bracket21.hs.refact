[("tests/examples/Bracket21.hs:1:1-35: Suggestion: Use newtype instead of data\nFound:\n  data Foo = Foo {foo :: (Maybe Foo)}\nPerhaps:\n  newtype Foo = Foo {foo :: (Maybe Foo)}\nNote: decreases laziness\n",[]),("tests/examples/Bracket21.hs:1:24-34: Suggestion: Redundant bracket\nFound:\n  foo :: (Maybe Foo)\nPerhaps:\n  foo :: Maybe Foo\n",[Replace {rtype = Type, pos = SrcSpan {startLine = 1, startCol = 24, endLine = 1, endCol = 35}, subts = [("x",SrcSpan {startLine = 1, startCol = 25, endLine = 1, endCol = 34})], orig = "x"}])]
