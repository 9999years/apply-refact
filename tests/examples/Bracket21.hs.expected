data Foo = Foo {foo :: Maybe Foo}
